src/sentry/api/endpoints/auth_index.py

```diff
@@ -75,9 +88,70 @@ def post(self, request):
                 {
                     'message': 'Cannot sign-in with basic auth because password has expired.',
                 },
-                status=403
+                status=status.HTTP_403_FORBIDDEN
             )
 
+        request.user = request._request.user
+
+        return self.get(request)
+
+    def put(self, request):
+        """
+        Verify a User
+        `````````````
+
+        This endpoint verifies the currently authenticated user (for example, to gain superuser).
+
+        :auth: required
+        """
+        if not request.user.is_authenticated():
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+        validator = AuthVerifyValidator(data=request.DATA)
+        if not validator.is_valid():
+            return self.respond(validator.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        authenticated = False
+
+        # See if we have a u2f challenge/response
+        if 'challenge' in validator.object and 'response' in validator.object:
+            try:
+                interface = Authenticator.objects.get_interface(request.user, 'u2f')
+                if not interface.is_enrolled:
+                    raise LookupError()
+
+                challenge = json.loads(validator.object['challenge'])
+                response = json.loads(validator.object['response'])
+                authenticated = interface.validate_response(request, challenge, response)
+            except ValueError:
+                pass
+            except LookupError:
+                pass
+
+        # attempt password authentication
+        else:
+            authenticated = request.user.check_password(validator.object['password'])
+
+        # UI treats 401s by redirecting, this 401 should be ignored
+        if not authenticated:
+            return Response({'detail': {'code': 'ignore'}}, status=status.HTTP_403_FORBIDDEN)
+
+        try:
+            # Must use the real request object that Django knows about
+            auth.login(request._request, request.user)
+            # Also can grant sudo
+            grant_sudo_privileges(request._request)
+        except auth.AuthUserPasswordExpired:
+            return Response(
+                {
+                    'code': 'password-expired',
+                    'message': 'Cannot sign-in with basic auth because password has expired.',
```

Comment:
Being pedantic: I wouldn't call this "basic auth" since that has a different meaning in HTTP with an actual Authorization header, not our session auth.